File: Data Structures & Algorithms/is-palindrome/submission-2.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        sent = [char for char in s if char.isalnum()]
        left = 0
        right = len(sent)-1
        while left < right:
            if sent[left] != sent[right]:
                return False
            left +=1
            right -=1
        return True

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        // sliding where the left would be left price, 
        // and we keep sliding and until we get to a lower price than the start, 
        // the left pointer becomes that
        int max_profit = 0;
        int left = prices[0];
        for (int price : prices) {
            if (left > price) {
                left = price;
            }
            max_profit = Math.max(max_profit, price-left);
        }
        return max_profit;
    }
}
